### ISSUE TITLE
[Bug] Using the search box then accessing a recipe or exiting the inventory, resets the search.

### ISSUE THREAD

**Issue by MechanosG** [shedaniel/RoughlyEnoughItems#442]
**Describe the bug**
Using the search box in the inventory, items and blocks on the right will filter correctly.  However, if you press R to check the recipe of anything, or exit the inventory screen, the filtered items and blocks reset as if you didn't search anything, despite the searched text still being in the search box.  You have to click the search box and make a change to the search query (such as deleting or adding a character) to get your results to filter again.  This can be frustrating when trying to craft things from a specific mod, only to lose your place repeatedly.

**Steps to Reproduce**
- Load a modpack containing RoughlyEnoughItems-5.8.5.jar (the last version I've tested)
- Open your inventory
- Type anything into the search box, for example "stone"
- You'll see the blocks and items on the right filter for stone
- Point at any of the results and press R, or just exit your inventory screen then re-open it.  The filtered results will have reset and went to page 1, making you lose your place.

**Screenshots**
![2020-11-16_19 52 28](https://user-images.githubusercontent.com/4360512/99344686-7a9e9600-2845-11eb-837e-40fd15b1f8f1.png)
In this first screenshot, I've searched "stone", and as you can see, stuff filtered correctly.

![2020-11-16_19 52 46](https://user-images.githubusercontent.com/4360512/99344697-8427fe00-2845-11eb-8820-7a45bbaa5a13.png)
I pressed R on vanilla stone to view the furnace recipe, and you'll notice the items and blocks on the right have already reset.

![2020-11-16_19 53 16](https://user-images.githubusercontent.com/4360512/99344714-8a1ddf00-2845-11eb-8675-328539554192.png)
Back in my inventory, the search box still contains "stone", yet the results are still reset as if the search box was empty, forcing me to use the search box again and change the string to get it to filter again.

**Environment (please complete the following information with the version):**
 - Minecraft: 1.16.3
 - Mod Loader: Fabric 0.6.1.51 via Jumploader 2.1.0

**Logs**
Probably not applicable, unless you need a mod list or something (conflict maybe).  I could get you a log if you want it.

**Additional context**
I know you're aware of the bug already since I commented about it on CurseForge and you replied, but I figured I'd go ahead and make an issue here, just in case it helps any.  Ty for REI :)

**Comment by shedaniel** [comment]
Are you able to reproduce with only REI, I was not able to during testing after I received that curseforge message

**Comment by justastranger** [comment]
I'm also experiencing this

**Comment by MechanosG** [comment]
Alright so, testing this is taking longer than I expected, so part of this will be a bit like a journal of the steps I'm taking as I take them, while parts will be my findings, so I don't forget what I'm doing and have already tested.  I'm not sure how to make spoilers in github either.

So first, I made a new MC install and tested REI by itself.  The bug didn't happen.  So on a hunch, I copied my config files from the main modpack into this install, then I installed Roughly Enough Resources, and 3 tech mods, Industrial Revolution, Modern Industrialization, and Tech Reborn, then repeated the test.  The bug immediately showed up.  So I temporarily disabled the 3 tech mods and their dependencies, so now I'm only loading Fabric API, Jumploader, REI, and Roughly Enough Resources, and repeated the test.  The bug is still showing up.  So it's either..  my REI config that I transferred over, or Roughly Enough Resources is causing the problem?  Eh, I'll delete my config folder and test just to rule it out - ... ok, unexpected.  Deleting my config folder made the bug go away, even with Roughly Enough Resources still active.

So I just ran the config files through WinMerge to compare the differences.  The only differences are, in the modpack config I accidentally favorited a poisonous potato at some point, I set favoritesEnabled to false so that I could use that screen space for a crafting mod, and I unbound the favorites hotkey.  I'm guessing this is most likely related to disabling the favorites, so I'll test that in the main modpack and in the bug test profile.

Testing the main modpack, the bug was present at first.  Enabling favorites made the bug go away.  But re-disabling the favorites, didn't bring the bug back while still in the same session.  Kinda pointing towards the bug happening only when favorites are disabled in the config when loading the game.  Sooo, I tested REI by itself, got in game, turned off favorites, saved and exited the game, reloaded the game, and..  finally, confirmation.

TL;DR - Yes, REI does it by itself.  Go in game, open inventory, disable favorites in REI, save and exit the game, reload the game and enter your world, open your inventory, put something in the search box, then look at a recipe or exit your inventory.  The bug shows itself.  Having favorites disabled when loading the game, seems to be what causes the problem.
